### ISSUE TITLE
Automatically create directories for the path passed to `new Database` 

### ISSUE THREAD

**Issue by ImLunaHey** [oven-sh/bun#3888]
### What version of Bun is running?

0.7.1

### What platform is your computer?

Darwin 21.6.0 x86_64 i386

### What steps can reproduce the bug?

- create a directory called `src`
- create a file called `a.ts` with this contents

    ```ts
    import { Database } from 'bun:sqlite';
    const db = new Database('./i-dont-exist/data.db');
    ```

- `bun run ./a.ts`
- error should show

### What is the expected behavior?

The directory to be created and then the new database file to be created.

### What do you see instead?

This error.

```bash
➜  paas git:(main) ✗ bun run src/service-registry.ts
136 |     if (anonymous && (flags & constants.SQLITE_OPEN_READONLY) !== 0)
137 |       throw new Error("Cannot open an anonymous database in read-only mode.");
138 |     if (!SQL)
139 |       _SQL = SQL = lazy("sqlite");
140 | 
141 |     this.#handle = SQL.open(anonymous ? ":memory:" : filename, flags), this.filename = filename;
                      ^
error: unable to open database file
      at new Database (bun:sqlite:141:19)
      at /Users/xo/code/ImLunaHey/paas/src/service-registry.ts:3:11
```

**Comment by ImLunaHey** [comment]
If I do the same but pass it a db that doesn't exist in the current directory the file is created.

**Comment by v3nntrnx** [comment]
I don't really think it's a bug. But we can implement this

**Comment by simylein** [comment]
You can pass `create: true` to the database constructor in its options. 

```ts
import { Database } from 'bun:sqlite';

new Database('./folder/database.sqlite', { create: true });
```

This will however not create missing directories, only missing database files.

**Comment by ImLunaHey** [comment]
> I don't really think it's a bug. But we can implement this

This feels like a bug, why exactly do I need to create the directory for the database to exist. This seems like pointless limitation, creating the directory would make the user experience better and honestly would do what most users expect.

@simylein even if that's not provided the file will be created on first write.

**Comment by ImLunaHey** [comment]
This is the same type of bug as https://github.com/oven-sh/bun/issues/1898

There's no good reason for needing a user to create a directory beforehand. Bun is more than capable of doing this itself.

**Comment by v3nntrnx** [comment]
@Jarred-Sumner should we auto create the directory here? (and/or when using in bun init, talking about the issue mentioned above )

**Comment by vjpr** [comment]
Potentially related: https://github.com/oven-sh/bun/issues/3863

`fs.open(path, <create flag>)` is broken.

**Comment by maxmilton** [comment]
Looking at the code in <https://github.com/oven-sh/bun/blob/main/src/js/bun/sqlite.ts#L168-L216>, it appears that bun is using the constructor options to set [bit flags](https://sqlite.org/c3ref/c_open_autoproxy.html) which are then passed to [sqlite3_open_v2](https://sqlite.org/c3ref/open.html). This means that it's actually SQLite that's currently responsible for creating the database file.

I think it's intuitive that the `create` option is the same as passing in `constants.SQLITE_OPEN_CREATE | constants.SQLITE_OPEN_READWRITE`. It's straightforward to create the directories yourself as well. All that said, I'm not opposed to automatically creating the directories, as long as it's done only when the create option is set to true and precautions are taken to prevent path traversal vulnerabilities etc. I never use `SQLITE_OPEN_CREATE` in production code anyway as I prefer to explicitly initialize databases separately.

**Comment by simylein** [comment]
I agree it should be explicit because otherwise you end up creating a new database in production and not noticing that you are not writing to your intended target file. More like explicit failure than silently creating a new file.

**Comment by truongan07** [comment]
i have more id-services on Microservices and i need to make directory is id-service in path to group all databases of this id-service. Ex: 
'./772a0175-58df-4b7a-a2c8-d53d62680d4f/db-1.sqlite3', 
'./772a0175-58df-4b7a-a2c8-d53d62680d4f/db-2.sqlite3'
